### PR TITLE
Add one more symint expand test.

### PR DESCRIPTION
**Pull request by vanbasten23**
Inspired by [GH issue](https://github.com/pytorch/xla/issues/4202), I'm adding another symint expand test in this PR.

### CODE REVIEW

**Comment by vanbasten23** [comment]
Right now, the test fails with `RuntimeError: SymIntArrayRef expected to contain only concrete integers`.

File: test/test_dynamic_shapes.py [hunk 1]
```diff
@@ -19,6 +19,18 @@ def test_simple_expand(self):
...
+  def test_simple_expand_on_2d_tensor(self):
+    size1 = 5
+    size2 = 2
+    t1 = torch.zeros([size1, size2], device=dev)
+    t1[3][0] = 1
+    t1[3][1] = 1
+    # t2 has size [<=10, 2]
+    t2 = torch.nonzero(t1)
+    t3 = torch.ones(1, size1, size2)
+    t4 = t3.expand(t2.shape[0], size1, size2)
```
Comment (JackCaoG) [inline test/test_dynamic_shapes.py:31]: @ezyang Does pytorch supports one of the expand dimension is dynamic and others are static?

**Comment by ezyang** [comment]
You can nail this down precisely if you get a c++ backtrace but usually this just means there isn't a _symint registration for the op (expand, I guess?)

**Comment by vanbasten23** [comment]
> You can nail this down precisely if you get a c++ backtrace but usually this just means there isn't a _symint registration for the op (expand, I guess?)

Thanks Ed. I've gotten the [c++ backtrace](https://gist.github.com/vanbasten23/a5b097ce557f2c0f01a7815aa4ea02b9). Can you show me where I can find the _symint registration for an op?

**Comment by JackCaoG** [comment]
@vanbasten23 Did you compile pytorch and pytorch/xla with `DEBUG=1`? A lot of the frame seems to be missing

**Comment by JackCaoG** [comment]
I expect the code to hit https://github.com/pytorch/xla/blob/653df694f7fafe5cda39fbc90e14e0347fde07ae/torch_xla/csrc/aten_xla_type.cpp#L1156-L1170

@vanbasten23 can you confirm?

**Comment by vanbasten23** [comment]
> @vanbasten23 can you confirm?

re @JackCaoG , ah I see where the problem is. In the current test, `t1`(`t1 = torch.zeros([size1, size2], device=dev)`) tensor is on xla device but `t3 = torch.ones(1, size1, size2)` is not. Perhaps that's why we see `frame #21: at::_ops::expand::call(at::Tensor const&, c10::ArrayRef<c10::SymInt>, bool) ` in the stacktrace, in which PyTorch dispatch the call to aten I guess?

Btw, I did compile pytorch and pt/xla with DEBUG=1 and that's all the stacktrace.

After I set the device for `t3` tensor to be XLA, the error goes away. Thanks for checking!

**Review by miladm — APPROVED** [review]
LGTM. Thanks.

**Comment by miladm** [comment]
UX question:
Turns out we got this `RuntimeError: SymIntArrayRef expected to contain only concrete integers` when developers don't have their tensors on the right device. Wondering if we can improve the error message quality to improve developer experience - before we hit [this line](https://github.com/pytorch/pytorch/blob/master/c10/core/SymIntArrayRef.h#L33-L36). @wconstab wdyt?

**Comment by wconstab** [comment]
> UX question:
Turns out we get this RuntimeError: SymIntArrayRef expected to contain only concrete integers when developers don't have their tensors on the right device. Wondering if we can improve the error message quality to improve developer experience - before we hit [this line](https://github.com/pytorch/pytorch/blob/master/c10/core/SymIntArrayRef.h#L33-L36). @wconstab wdyt?

What did you have in mind @miladm?  That runtime check you linked is really general- someone asked to convert a symint arrayref to an intarrayref, and we're safety checking that none of them are indeed symbolic (which would make the conversion an error).  This conversion gets called from all over.  It's not guaranteed that the issue is due to not using the XLA device.

**Comment by ezyang** [comment]
this PR shoudl help https://github.com/pytorch/pytorch/pull/90675 someone please review it lol

**Comment by ezyang** [comment]
this PR shoudl help https://github.com/pytorch/pytorch/pull/90675 someone please review it lol

**Comment by miladm** [comment]
Awesome! Thanks @ezyang.

**Pull request merged**